### PR TITLE
Update base map developer guide

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- Using with Mapbox: https://pessimistress.github.io/deck.gl/docs/developer-guide/base-maps/using-with-mapbox
  + cc @jonenst 
  + Reorganized for clarity, coined new term *reverse controlled* to replace "inserting Mapbox into the deck.gl container"
  + Removed sections about forks and self hosting, some of the content got merged in to the Maplibre page
- New page: using with Maplibre https://pessimistress.github.io/deck.gl/docs/developer-guide/base-maps/using-with-maplibre
  + cc @birkskyum @HarelM
  + The first half is mostly the same as the Mapbox page with some find/replace
  + Missing a hero screenshot here - the Mapbox one contains Mapbox building data so we probably do not want to reuse
- Using with Google Maps: https://pessimistress.github.io/deck.gl/docs/developer-guide/base-maps/using-with-google-maps
  + cc @usefulthink
  + Added React examples using react-google-maps 
- MapboxOverlay and GoogleMapsOverlay API references are updated with the same examples from the developer guide

### CODE REVIEW

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+# Using with Maplibre
+
+| Pure JS | React | Overlaid | Interleaved |
+| ----- | ----- | ----- | ----- |
+|  ✓ | ✓ | [example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/maplibre) | [example](https://deck.gl/gallery/maplibre-overlay) |
+
+[Maplibre GL JS](https://maplibre.org/maplibre-gl-js/docs/) is a powerful open-source library that uses WebGL to render interactive maps from vector tiles in a browser. deck.gl's `MapView` can sync perfectly with the camera of Maplibre, at every zoom level and rotation angle.
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:7]: In mapbox there's a nice looking image in this part, can maplibre has a nice looking image too?

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+## Integration Modes
+
+When using deck.gl and Maplibre, there are three options you can choose from: interleaved, overlaid, and reverse-controlled.
+
+### Interleaved
+
+The [interleaved](../../get-started/using-with-map.md#interleaved) mode renders deck.gl layers into the WebGL2 context created by Maplibre. If you need to mix deck.gl layers with Maplibre layers, e.g. having deck.gl surfaces below text labels, or objects occluding each other correctly in 3D, then you have to use this option.
+
+Interleaving is supported by using [MapboxOverlay](../../api-reference/mapbox/mapbox-overlay.md) with `interleaved: true`. It requires WebGL2 and therefore only works with `maplibre-gl@>3`. See [compatibility](../../api-reference/mapbox/overview#interleaved-renderer-compatibility) and [limitations](../../api-reference/mapbox/overview.md#limitations).
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:17]: You know what I think about the name "mapbox" in both the npm package and the class name so I won't repeat it here.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+- [TileZen schema](https://tilezen.readthedocs.io/en/latest/layers/)
+- [OpenMapTiles schema ](https://openmaptiles.org/schema/)
+- [Mapbox Vector Tile Spec](https://www.mapbox.com/developers/vector-tiles/)
+
+Some useful resources for creating your own map service:
+
+- [Martin](https://github.com/maplibre/martin), a tile server from the Maplibre organization
+- [Open source tools](https://github.com/mapbox/awesome-vector-tiles)
+- [Maputnik Style editor](https://maputnik.github.io)
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:248]:
```suggestion
- [Maputnik Style editor](https://maplibre.org/maputnik/)
```

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+All the [examples on this website](https://github.com/visgl/deck.gl/tree/master/examples/website) are implemented using the React integration.
+
+When you choose the interleaved or overlaid option, the react-map-gl [Map](https://visgl.github.io/react-map-gl/docs/api-reference/map) React component acts as the root component, and [MapboxOverlay](../../api-reference/mapbox/mapbox-overlay#using-with-react-map-gl) is used with react-map-gl's `useControl` hook. 
+
+When you choose the reverse-controlled option, the `DeckGL` React component cats as the root component, and the react-map-gl [Map](https://visgl.github.io/react-map-gl/docs/api-reference/map) is a child. In this case, `Map` will automatically interpret the deck.gl view state (i.e. latitude, longitude, zoom etc), so that deck.gl layers will render as a synchronized geospatial overlay over the underlying map.
+
+
+### Choosing a map tile service
+
+There are paid map tile servers such as [MapTiler](https://www.maptiler.com/) and [AWS Location Service](https://docs.aws.amazon.com/location/latest/developerguide/map-concepts.html).
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:232]: There are more tiles providers/servers. I'm not sure which are entitled to be here, but stadia maps comes to mind.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+### Example: reverse controlled
+
+The reverse-controlled option is supported by the pre-built scripting bundle, and in React when used with the `react-map-gl` library. There is currently no easy way to do it under Vanilla JS.
+
+<Tabs groupId="language">
+  <TabItem value="scripting" label="Scripting">
+
+```js
+<script src="https://unpkg.com/deck.gl@^9.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/maplibre-gl@^3.0.0/dist/maplibre-gl.js"></script>
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:143]: Consider linking to version 4, I don't know if this was tested or not, but I think it's better to link to the latest version if possible.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+<link href="https://unpkg.com/maplibre-gl@^3.0.0/dist/maplibre-gl.css" rel="stylesheet" />
+<script type="text/javascript">
+  const {DeckGL, ScatterplotLayer} = deck;
+
+  maplibregl.accessToken = '<mapbox_access_token>';
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:148]: `accessToken` is no longer supported without a plugin.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs groupId="language">
+  <TabItem value="ts" label="TypeScript">
+
+```ts
+import {MapboxOverlay} from '@deck.gl/mapbox';
+import {ScatterplotLayer} from '@deck.gl/layers';
+import maplibregl from 'maplibre-gl';
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:50]: As an example, I think importing the `Map` object is a better example of how to use maplibre-gl.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+import 'maplibre-gl/dist/maplibre-gl.css';
+
+const map = new maplibregl.Map({
+  container: 'map',
+  style: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+  center: [0.45, 51.47],
+  zoom: 11
+});
+
+map.once('load', () => {
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:60]: `once` returns a promise, consider changing it to use `await map.once('load')`.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+
+You can find full project setups in the [react get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/react/maplibre/) and [pure js get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/maplibre/).
+
+
+### Example: reverse controlled
+
+The reverse-controlled option is supported by the pre-built scripting bundle, and in React when used with the `react-map-gl` library. There is currently no easy way to do it under Vanilla JS.
+
+<Tabs groupId="language">
+  <TabItem value="scripting" label="Scripting">
```
Comment (HarelM) [inline docs/developer-guide/base-maps/using-with-maplibre.md:139]: `scripting` is an odd decision from my point of view when reading the docs.
Consider changing it to HTML, or JavaScript.

**Comment by HarelM** [comment]
I've added my nit-picking comments.
Thanks for cc'ing me and for this great write up!

File: docs/developer-guide/base-maps/using-with-maplibre.md (hunk 1 above)
Comment (Pessimistress) [reply docs/developer-guide/base-maps/using-with-maplibre.md:7]: Any suggestions? The Mapbox one contains building/style from the Mapbox data service, so I didn't want to reuse it. Anything new features worth showcasing of Maplibre?

File: docs/developer-guide/base-maps/using-with-maplibre.md (hunk 8 above)
Comment (Pessimistress) [reply docs/developer-guide/base-maps/using-with-maplibre.md:60]: I didn't know this. It's very nice indeed.

File: docs/developer-guide/base-maps/using-with-maplibre.md (hunk 1 above)
Comment (HarelM) [reply docs/developer-guide/base-maps/using-with-maplibre.md:7]: I looked at the deck.gl examples in maplibre, but they don't create any wow effect.
https://maplibre.org/maplibre-gl-js/docs/examples/toggle-deckgl-layer/
https://maplibre.org/maplibre-gl-js/docs/examples/add-deckgl-layer-using-rest-api/

But I think any picture is better than no picture, feel free to take a picture from any of the above examples.

File: docs/developer-guide/base-maps/using-with-maplibre.md [hunk 10]
```diff
@@ -0,0 +1,246 @@
+# Using with Maplibre
```
Comment (birkskyum) [inline docs/developer-guide/base-maps/using-with-maplibre.md:1]: Small nit, we always spell MapLibre with a capital L. It's a trademark we've registered, so we try to be consistent.

File: docs/developer-guide/base-maps/using-with-maplibre.md (hunk 9 above)
Comment (Pessimistress) [reply docs/developer-guide/base-maps/using-with-maplibre.md:139]: I admit it's a dated term, though `HTML` kind of implies that it's used *with* the TypeScript code, not an alternative to it?

File: docs/developer-guide/base-maps/using-with-google-maps.md [hunk 11]
```diff
@@ -6,24 +6,187 @@
...
+
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs groupId="language">
+  <TabItem value="ts" label="TypeScript">
+
+```ts
+import {Loader} from "@googlemaps/js-api-loader";
```
Comment (usefulthink) [inline docs/developer-guide/base-maps/using-with-google-maps.md:47]:
```suggestion
import {Loader} from '@googlemaps/js-api-loader';
```

File: docs/developer-guide/base-maps/using-with-google-maps.md [hunk 12]
```diff
@@ -6,24 +6,187 @@
...
+import {GoogleMapsOverlay} from '@deck.gl/google-maps';
+import {ScatterplotLayer} from '@deck.gl/layers';
+
+const loader = new Loader({
+  apiKey: '<google_maps_api_key>',
+  version: "quarterly"
+});
```
Comment (usefulthink) [inline docs/developer-guide/base-maps/using-with-google-maps.md:54]: As above, I think we should leave the version out or use the default here:
```suggestion
const loader = new Loader({apiKey: '<google_maps_api_key>'});
```

...or...

```suggestion
const loader = new Loader({
  apiKey: '<google_maps_api_key>',
  version: 'weekly'
});
```

File: docs/developer-guide/base-maps/using-with-google-maps.md [hunk 13]
```diff
@@ -6,24 +6,187 @@
...
+  const overlay = useMemo(() => new GoogleMapsOverlay(props));
 
-## Using overlaid
+  useEffect(() => {
+    if (map) {
+      overlay.setMap(map);
+      return () => overlay.setMap(null);
+    }
+    return undefined;
+  }, [map])
```
Comment (usefulthink) [inline docs/developer-guide/base-maps/using-with-google-maps.md:101]: Same as above
```suggestion
  useEffect(() => {
    overlay.setMap(map);
    
    return () => overlay.setMap(null);
  }, [map]);
```

**Review by usefulthink — COMMENTED** [review]
Excellent summary, thanks a lot! I added a couple of smaller suggestions, nothing major.

File: docs/developer-guide/base-maps/using-with-mapbox.md [hunk 14]
```diff
@@ -6,144 +6,249 @@
...
-</Tabs>
+You can find full project setups in the [react get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/react/mapbox/) and [pure js get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/mapbox/).
 
-If you want to use features of mapbox-gl, such as controls like `NavigationControl` or plugins, then need to insert Deck into the Mapbox map container using [MapboxOverlay](../../api-reference/mapbox/mapbox-overlay) from the [@deck.gl/mapbox](../../api-reference/mapbox/overview.md) module, which allows you to construct a Deck instance and apply it to a map using the [IControl](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol) API. In this configuration mapbox-gl handles all user input, holds the source of truth of the camera state, and is the root HTML element of your map.
 
-There are two renderers to pick from, overlaid or interleaved.
+### Example: reverse controlled
 
-### Overlaid
+The reverse-controlled option is supported by the pre-built scripting bundle, and in React when used with the `react-map-gl` library. There is currently no easy way to do it under Vanilla JS.
```
Comment (chrisgervang) [inline docs/developer-guide/base-maps/using-with-mapbox.md:140]: Nice! This is clearer than before.

**Review by chrisgervang — APPROVED** [review]

**Pull request merged**